### DISCUSSION TITLE
Raspberry Pi 3 B+ compatibility

### DISCUSSION THREAD

**Discussion by moodkillah** [fayaaz/mixxx-pi-gen#33] [Q&A] [answered]
HI, im using a raspberry pi 3 b+ but cant seem to boot, is this image compatible with my board?

**Comment by fayaaz** [comment]
The recent versions are 64 bit so only compatible with RPi4.

You can try the version with 2.2 here: https://github.com/fayaaz/mixxx-pi-gen/releases/tag/v0.1.0

Unfortunately I cannot see much gain in maintaining the armhf version as the performance on a 3b is too slow to be worthwhile.

**Comment by moodkillah** [comment]
i only use mixxx for 1 deck, so i dont stress to much the pi3 b+. 
i wanted to try mixxxx 2.3 on my pi 3 but im having problems compiling on raspbian, so maybe i should wait for the official repositories update so i can sudo apt update mixxxx .

**Reply by fayaaz** [reply]
I've actually ended up down a rabbit hole and it turns out the pi3 b+ should be 64 bit but it requires some non-free firmware blob to boot potentially? Anyway I've made a small change in this PR: https://github.com/fayaaz/mixxx-pi-gen/pull/35

The actions on that should build an armhf version for the deb and the image - it may be worth seeing if that works for you

**Reply by JaccoR** [reply]
It is not working for me on a raspberry pi 3b+. What do you mean by non-free firmware blob?

**Reply by fayaaz** [reply]
@JaccoR are you able to boot the 64-bit minimal version of Raspberry Pi OS on your 3b+?

**Reply by JaccoR** [reply]
Where can i find this version?

**Reply by fayaaz** [reply]
https://downloads.raspberrypi.org/raspios_arm64/

**Reply by gueritta** [reply]
Just tried this, and the minimal version from the link above boots.

**Comment by fayaaz** [comment]
Is anyone with a 3B - @JaccoR @servleto @moodkillah able to test this image: https://github.com/fayaaz/mixxx-pi-gen/suites/5032642210/artifacts/148631835

It is now updated to Debian 11 so should be similar to the working minimal version.

**Reply by alhadebe** [reply]
@fayaaz I tried this image and it did not work. What you need and how do i get to debug?

**Reply by fayaaz** [reply]
> @fayaaz I tried this image and it did not work. What you need and how do i get to debug?

Did it at least boot to any graphical interface or a terminal or just a black screen?

**Reply by alhadebe** [reply]
Boots to a terminal with text.

**Reply by fayaaz** [reply]
Ok that's something @NotYourAverageAl - would you be able to grab the logs from /var/log/  - specifically useful are:
- `/var/log/boot.log`
- `/var/log/Xorg.0.log`
- `/var/log/lightdm/lightdm.log`

If you are booting to a terminal it means that at least we are getting into the OS but crashing at the graphical target.

One last thing is to check your GPU memory split using `sudo raspi-config` in the terminal. You'll need at least 128 MB I expect.

**Comment by fayaaz** [comment, marked as answer]
Right all the Raspberry Pi 3 people @NotYourAverageAl,  @JaccoR, @servleto, @moodkillah I have managed to test out the image on my RPi3 which was being used for something else until recently.

The problem is the `threadirqs` parameter on the kernel. This causes the Pi 3Bs to kernel panic at startup. Not sure why it works on the Pi4 but not the 3s but anyway the fix is easy enough. Once you flash the image open up `/boot/cmdline.txt` on whatever PC  you used to do the flash, and remove the `threadirqs` part from the end of the line. Do not remove the whole line.
Mine looks like this:

```
console=serial0,115200 console=tty1 root=ROOTDEV rootfstype=ext4 fsck.repair=yes rootwait quiet init=/usr/lib/raspi-config/init_resize.sh
```


I have asked in Mixxx's zulip chat about `threadirqs` and if it is needed any more. There is an ongoing discussion here:
https://mixxx.zulipchat.com/#narrow/stream/109122-general/topic/Realtime.20.2F.20low.20latency.3A.20threadirqs.20vs.20PREEMPT_RT.20patches
